File: taurus-protect-sdk-java/client/src/main/java/com/taurushq/sdk/protect/client/model/Group.java
package com.taurushq.sdk.protect.client.model;

import java.time.OffsetDateTime;
import java.util.List;

/**
 * Represents a user group in the Taurus Protect system.
 * <p>
 * Groups are used to organize users and define access permissions
 * and approval workflows.
 *
 * @see GroupService
 */
public class Group {

    private String id;
    private String tenantId;
    private String externalGroupId;
    private String name;
    private String email;
    private List<GroupUser> users;
    private OffsetDateTime creationDate;
    private OffsetDateTime updateDate;
    private String description;
    private Boolean enforcedInRules;

    /**
     * Gets the unique identifier of the group.
     *
     * @return the group ID
     */
    public String getId() {
        return id;
    }

    /**
     * Sets the group ID.
     *
     * @param id the ID to set
     */
    public void setId(String id) {
        this.id = id;
    }

    /**
     * Gets the tenant ID this group belongs to.
     *
     * @return the tenant ID
     */
    public String getTenantId() {
        return tenantId;
    }

    /**
     * Sets the tenant ID.
     *
     * @param tenantId the tenant ID to set
     */
    public void setTenantId(String tenantId) {
        this.tenantId = tenantId;
    }

    /**
     * Gets the external group ID (e.g., from an identity provider).
     *
     * @return the external group ID
     */
    public String getExternalGroupId() {
        return externalGroupId;
    }

    /**
     * Sets the external group ID.
     *
     * @param externalGroupId the external group ID to set
     */
    public void setExternalGroupId(String externalGroupId) {
        this.externalGroupId = externalGroupId;
    }

    /**
     * Gets the group name.
     *
     * @return the group name
     */
    public String getName() {
        return name;
    }

    /**
     * Sets the group name.
     *
     * @param name the name to set
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * Gets the group email address.
     *
     * @return the group email
     */
    public String getEmail() {
        return email;
    }

    /**
     * Sets the group email.
     *
     * @param email the email to set
     */
    public void setEmail(String email) {
        this.email = email;
    }

    /**
     * Gets the list of users in this group.
     *
     * @return the list of group users
     */
    public List<GroupUser> getUsers() {
        return users;
    }

    /**
     * Sets the list of users.
     *
     * @param users the users to set
     */
    public void setUsers(List<GroupUser> users) {
        this.users = users;
    }

    /**
     * Gets the creation date of the group.
     *
     * @return the creation date
     */
    public OffsetDateTime getCreationDate() {
        return creationDate;
    }

    /**
     * Sets the creation date.
     *
     * @param creationDate the creation date to set
     */
    public void setCreationDate(OffsetDateTime creationDate) {
        this.creationDate = creationDate;
    }

    /**
     * Gets the last update date of the group.
     *
     * @return the update date
     */
    public OffsetDateTime getUpdateDate() {
        return updateDate;
    }

    /**
     * Sets the update date.
     *
     * @param updateDate the update date to set
     */
    public void setUpdateDate(OffsetDateTime updateDate) {
        this.updateDate = updateDate;
    }

    /**
     * Gets the group description.
     *
     * @return the description
     */
    public String getDescription() {
        return description;
    }

    /**
     * Sets the description.
     *
     * @param description the description to set
     */
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * Returns whether this group is enforced in business rules.
     *
     * @return true if enforced in rules, false otherwise
     */
    public Boolean getEnforcedInRules() {
        return enforcedInRules;
    }

    /**
     * Sets whether this group is enforced in rules.
     *
     * @param enforcedInRules the enforcement flag
     */
    public void setEnforcedInRules(Boolean enforcedInRules) {
        this.enforcedInRules = enforcedInRules;
    }
}

File: taurus-protect-sdk-go/pkg/protect/model/group.go
package model

import "time"

// Group represents a user group in the system.
type Group struct {
	// ID is the unique identifier for the group.
	ID string `json:"id"`
	// TenantID is the tenant this group belongs to.
	TenantID string `json:"tenant_id,omitempty"`
	// ExternalGroupID is the external identifier for the group.
	ExternalGroupID string `json:"external_group_id,omitempty"`
	// Name is the name of the group.
	Name string `json:"name"`
	// Email is the email address associated with the group.
	Email string `json:"email,omitempty"`
	// Description is a description of the group.
	Description string `json:"description,omitempty"`
	// EnforcedInRules indicates if the group is enforced in rules.
	EnforcedInRules bool `json:"enforced_in_rules"`
	// Users is the list of users in the group.
	Users []GroupUser `json:"users,omitempty"`
	// CreatedAt is when the group was created.
	CreatedAt time.Time `json:"created_at"`
	// UpdatedAt is when the group was last updated.
	UpdatedAt time.Time `json:"updated_at"`
}

// GroupUser represents a user within a group.
type GroupUser struct {
	// ID is the unique identifier for the user.
	ID string `json:"id"`
	// ExternalUserID is the external identifier for the user.
	ExternalUserID string `json:"external_user_id,omitempty"`
	// EnforcedInRules indicates if the user is enforced in rules.
	EnforcedInRules bool `json:"enforced_in_rules"`
}

// ListGroupsOptions contains options for listing groups.
type ListGroupsOptions struct {
	// Limit is the maximum number of groups to return.
	Limit int64
	// Offset is the number of groups to skip.
	Offset int64
	// IDs filters by group IDs.
	IDs []string
	// ExternalGroupIDs filters by external group IDs.
	ExternalGroupIDs []string
	// Query is a search query to filter groups.
	Query string
}

// ListGroupsResult contains the result of listing groups.
type ListGroupsResult struct {
	// Groups is the list of groups.
	Groups []*Group `json:"groups"`
	// TotalItems is the total number of groups matching the filter.
	TotalItems int64 `json:"total_items"`
	// Offset is the offset used for pagination.
	Offset int64 `json:"offset"`
}
